homeassistant/components/totalconnect/binary_sensor.py

```diff
@@ -14,30 +19,60 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up TotalConnect device sensors based on a config entry."""
-    sensors = []
+    sensors: Any = []
 
     client_locations = hass.data[DOMAIN][entry.entry_id].client.locations
 
     for location_id, location in client_locations.items():
+
+        sensors.append(TotalConnectAlarmLowBatteryBinarySensor(location))
+        sensors.append(TotalConnectAlarmTamperBinarySensor(location))
+        sensors.append(TotalConnectAlarmPowerBinarySensor(location))
+
         for zone_id, zone in location.zones.items():
-            sensors.append(TotalConnectBinarySensor(zone_id, location_id, zone))
+            sensors.append(TotalConnectZoneBinarySensor(zone_id, location_id, zone))
+
+            if not zone.is_type_button():
+                sensors.append(
+                    TotalConnectLowBatteryBinarySensor(zone_id, location_id, zone)
+                )
+                sensors.append(
+                    TotalConnectTamperBinarySensor(zone_id, location_id, zone)
+                )
 
     async_add_entities(sensors, True)
 
 
+class TotalConnectLowBatteryDescription(BinarySensorEntityDescription):
+    """EntityDescription for low battery."""
+
+    device_class = BinarySensorDeviceClass.BATTERY
+    entity_category = EntityCategory.DIAGNOSTIC
+
+
+class TotalConnectTamperDescription(BinarySensorEntityDescription):
+    """EntityDescription for tamper."""
+
+    device_class = BinarySensorDeviceClass.TAMPER
+    entity_category = EntityCategory.DIAGNOSTIC
+
+
+class TotalConnectPowerDescription(BinarySensorEntityDescription):
+    """EntityDescription for power sensor."""
+
+    device_class = BinarySensorDeviceClass.POWER
+    entity_category = EntityCategory.DIAGNOSTIC
+
+
 class TotalConnectBinarySensor(BinarySensorEntity):
     """Represent an TotalConnect zone."""
 
-    def __init__(self, zone_id, location_id, zone):
-        """Initialize the TotalConnect status."""
-        self._zone_id = zone_id
-        self._location_id = location_id
-        self._zone = zone
-        self._name = self._zone.description
-        self._unique_id = f"{location_id} {zone_id}"
-        self._is_on = None
-        self._is_tampered = None
-        self._is_low_battery = None
+    _unique_id = None
+    _name = None
+    _zone = None
+    _zone_id = None
+    _location_id = None
+    _is_on = None
```

Comment:
While you're here cleaning things up, we have built int attribute -> property mappings in HA for `_attr_name`, `_attr_unique_id`, `_attr_is_on`
```suggestion
    _zone = None
    _zone_id = None
    _location_id = None
```